tensorflow/contrib/rnn/BUILD

```diff
@@ -8,12 +8,37 @@ exports_files(["LICENSE"])
 
 package(default_visibility = ["//tensorflow:__subpackages__"])
 
+load("//tensorflow:tensorflow.bzl", "tf_custom_op_library")
+load("//tensorflow:tensorflow.bzl", "tf_gen_op_libs")
+load("//tensorflow:tensorflow.bzl", "tf_gen_op_wrapper_py")
 load("//tensorflow:tensorflow.bzl", "cuda_py_tests")
 
+tf_custom_op_library(
+    name = "python/ops/_lstm_ops.so",
+    srcs = [
+        "ops/lstm_ops.cc",
+    ],
+    hdrs = ["kernels/lstm_ops.h"],
+    gpu_srcs = [
+        "kernels/lstm_ops.cc",
```

Comment:
nvm my last comment, i think i can get it to compile... not familiar enough
w/ blaze BUILD files, not clear to me how its able to automatically pickup
the header even tho we don't specify it in the cc_binary... fixing other
issues atm.

--
William Chan
Carnegie Mellon University
(650) 450-9455
williamchan.ca

On Thu, Apr 28, 2016 at 1:24 PM, William Chan <williamchan@cmu.edu> wrote:

> where am i supposed to put the kernels/lstm_ops.cc ? srces doesn't work
> because it is used to build a binary which doesn't allow a header entry
> needed for the kernels/lstm_ops.cc?
>
> that's why i suggested earlier to refactor the tf_custom_op_library and
> into a tf_custom_kernel_library which builds two seperate cc_library, one
> for _gpu and one for the cpu case? isn't that the more correct approach?
>
> --
> William Chan
> Carnegie Mellon University
> (650) 450-9455
> williamchan.ca
>
> On Thu, Apr 28, 2016 at 1:01 PM, Manjunath Kudlur <
> notifications@github.com> wrote:
>
>> In tensorflow/contrib/rnn/BUILD
>> <https://github.com/tensorflow/tensorflow/pull/2002#discussion_r61492761>
>> :
>>
>> >  load("//tensorflow:tensorflow.bzl", "cuda_py_tests")
>> >
>> > +tf_custom_op_library(
>> > +    name = "python/ops/_lstm_ops.so",
>> > +    srcs = [
>> > +        "ops/lstm_ops.cc",
>> > +    ],
>> > +    hdrs = ["kernels/lstm_ops.h"],
>> > +    gpu_srcs = [
>> > +        "kernels/lstm_ops.cc",
>>
>> Nope. Only lstm_ops_gpu.cu.cc should go here.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/tensorflow/tensorflow/pull/2002/files/36fb73b93c217d59a868524383b347104ce431dd#r61492761>
>>
>
>